### PR TITLE
Kk/auto generate api reference

### PR DESCRIPTION
**Pull request by krzkaczor**

### CODE REVIEW

**Comment by hasparus** [comment]
@krzkaczor could you take a look at this? There's still the case of encoding special characters, but it's some progress. I'll test it more thoroughly tomorrow morning, but I suppose it's ready for review.

File: packages/api-reference-generator/src/utils.ts [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+    .replace(/[()\.,:\[\]]/g, '')
+    .replace(/<(.*?)>/g, '')
+    .replace(/ /g, '-')
+}
+
+export function encodeLabel(input: string): string {
+  return input.replace(/</g, '\\<')
+}
+
+// @todo
```
Comment (krzkaczor) [inline packages/api-reference-generator/src/utils.ts:14]: feel free to remove these comments 😆

File: packages/api-reference-generator/CHANGELOG.md [hunk 2]
```diff
@@ -0,0 +1,91 @@
+# earljs
```
Comment (krzkaczor) [inline packages/api-reference-generator/CHANGELOG.md:1]: this file was committed by accident i think

File: packages/docs/api/api-reference.md [hunk 3]
```diff
@@ -1,627 +1,551 @@
 ---
 id: api-reference
-title: API reference
+hide_title: true
```
Comment (krzkaczor) [inline packages/docs/api/api-reference.md:3]: I get that it makes sense to have this file committed right now to make review process simpler but i think when we are done we should gitignore this file and generate it on the fly when building website

File: packages/docs/api/api-reference.md [hunk 4]
```diff
@@ -1,627 +1,551 @@
...
+#### **`not(): Expectation<T>`**
 
-You can use matchers in place of a value. When a matcher is encountered its
-internal rules are used instead of the usual checks.
+* Inverts the behaviour of the validator that follows.
+  
+#### **`toBeA(this: Expectation<T>, clazz: any): void`**
 
-Arguments:
+* Checks if the value is an instance of the provided class or primitive type. Examples:
```
Comment (krzkaczor) [inline packages/docs/api/api-reference.md:83]: I am not sure why but there is a bullet point here 🤔

**Review by krzkaczor — COMMENTED** [review]
Good progress! Still, there is some work to do on the readability of the output (see images, the former is so much easier to skim) but definitely, we are getting closer!

One note is that I think we will need to add a little bit of handcrafted configuration to the generator. For example `not` is a modifier, not a validator, and should be listed as such.

Previous:
image

Current:
image

**Comment by hasparus** [comment]
We've reached the limit of parsing files by regex — I added a few hacks to make the API reference look good. I'm unhappy with it, but I suppose it's fine for now.

**Comment by hasparus** [comment]
@krzkaczor I deployed this branch here: https://earljs-docs-hasparus1.vercel.app/docs/api/api-reference

**Pull request merged**